src/main/java/mel/tasks/Deadline.java

```diff
@@ -0,0 +1,37 @@
+package mel.tasks;
+
+import mel.exceptions.ParseException;
+import mel.exceptions.TaskException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline class that represents a task with a deadlines.
+ */
+public class Deadline extends Task {
+    private final LocalDateTime by;
+
+    /**
+     * Constructs new Deadline task.
+     * @param input task input with a deadline, in the format:
+     *              deadline *task* /by *date* *time*.
+     * @throws TaskException if input is of invalid format.
+     */
+    public Deadline(String input) throws TaskException {
+        try {
+            String[] temp = input.split("/by ");
+            task = temp[0].split(" ", 2)[1];
+            by = super.parseDateTime(temp[1]);
+        } catch (ArrayIndexOutOfBoundsException | ParseException e) {
```

Comment:
Would having separate error messages for the separate exceptions be clearer?

Reply:
I suppose so, although I treated them similarly here because both `ArrayIndexOutOfBoundsException` and `ParseException` would produce the same effect of incorrect formatting causing a parse error. Thus, I opted to throw the same error message with `ParseException` in this case. Thank you! 😄